### ISSUE TITLE
Temporal consistency

### ISSUE THREAD

**Issue by nekoshadow1** [apple/ml-depth-pro#1]
Thank you for this amazing work! Can the model be applied to each frame of video? If so, how do you maintain temporal consistency?

**Comment by acgourley** [comment]
I'm interested in this too! It makes sense that a mono depth model would struggle with absolute scale across frames, but if anyone has had luck grafting on temporal consistency with another algorithm, or any authors know of plans to extend the core models to do this, I would like to know.

**Comment by elvistheyo** [comment]
Hi @nekoshadow1 were you able to find out how to maintain temporal consistency?

**Comment by nekoshadow1** [comment]
> Hi @nekoshadow1 were you able to find out how to maintain temporal consistency?

No, I have tested the model on video. I'm afraid the model cannot maintain temporal consistency between frames since I can observe flickers.  They are not severe though.

**Comment by elvistheyo** [comment]
@nekoshadow1 would you happen to know any method to overcome this flickering?

**Comment by nekoshadow1** [comment]
> @nekoshadow1 would you happen to know any method to overcome this flickering?

You may try [DepthCrafter](https://github.com/Tencent/DepthCrafter) or [Depth Any Video](https://github.com/Nightmare-n/DepthAnyVideo), which are specifically designed to estimate depth for video.

**Comment by CDchenlin** [comment]
> > @nekoshadow1 would you happen to know any method to overcome this flickering?
> 
> You may try [DepthCrafter](https://github.com/Tencent/DepthCrafter) or [Depth Any Video](https://github.com/Nightmare-n/DepthAnyVideo), which are specifically designed to estimate depth for video.

I actually doubt whether DepthCrafter or Depth Any Video are suitable for most video scenarios, as they cannot handle online prediction.

**Comment by elvistheyo** [comment]
> > @nekoshadow1 would you happen to know any method to overcome this flickering?
> 
> You may try [DepthCrafter](https://github.com/Tencent/DepthCrafter) or [Depth Any Video](https://github.com/Nightmare-n/DepthAnyVideo), which are specifically designed to estimate depth for video.

@nekoshadow1 instead of other models, is there any method that could improve the temporal consistency for depth pro directly?

**Comment by hbardak** [comment]
Temporal consistency would be amazing to have 
Here a test with DepthCrafter vs Depth Pro 

I have specified the focal length of my Camera into Depth pro into model.infer()


https://github.com/user-attachments/assets/e417c828-9f36-43cf-9580-79ff8b5aae4f

If only will could combine Depth pro's accuracy with DepthCrafter consistency !

**Comment by elvistheyo** [comment]
@hbardak what are you using to visualize the point clouds like a video?

**Comment by hbardak** [comment]
I used Nuke from The Foundry. It is a Compositing software to do VFX
